### ISSUE TITLE
`Dropout()` and `mlp()` in decoders

### ISSUE THREAD

**Issue by Newiz430** [EdisonLeeeee/MaskGAE#4]
Dear authors,

You've done an awesome job in MaskMAE! 
I have a question about the implementation of `DegreeDecoder`.

The `self.dropout(x)` (model.py line 300) is placed AFTER `mlp(x)`, while it is placed BEFORE `mlp(x)` in `EdgeDecoder`.
I can get "Final Test: 83.48 ± 0.27" by directly running `python train_nodeclas.py --dataset Cora --bn --l2_normalize --alpha 0.004` on my machine.
When I swapped two lines above in `DegreeDecoder.forward()` and ran the same command, I got "Final Test: 82.23 ± 0.15" which indicated a serious performance gap.
I've tried seed 2023, where I got mlp+dropout: 82.61 ± 0.28, dropout+mlp: 82.38 ± 0.66. The gap still exists. 

So what's the actual difference between mlp+dropout and dropout+mlp? Why the difference setting in `EdgeDecoder` and `DegreeDecoder`?
Thanks!

**Comment by EdisonLeeeee** [comment]
Thank you for your interest to our work!

The choice between the two architectures depends on the specific problem and the behavior of the model during training. MLP+dropout can be effective when the model is prone to overfitting (particularly for the simple degree regression task) and relies too heavily on a small subset of features. On the other hand, dropout+MLP can be effective when the input features are noisy or when there is a large number of input features.

As the dataset size increases (e.g., arXiv and MAG), the difference between MLP+dropout and dropout+MLP architectures may become less significant.

**Comment by Newiz430** [comment]
@EdisonLeeeee Thank you for reply. This helps a lot!
